src/OpenLoco/GameCommands/ChangeLoan.cpp

```diff
@@ -1,5 +1,6 @@
 #include "../CompanyManager.h"
 #include "../Economy/Economy.h"
+#include "../GameState.h"
```

Comment:
One thing i'm trying to avoid is having gamestate included in every file. The problem is that gamestate will be one of the heaviest headers we have and will very often have changes. We don't really want to force an absolute ton of recompiles. I was sort of thinking we would make it so that manager files include gamestate.h and provide getters/setters from there.

Reply:
Is compile time really a problem in open loco though? I feel like adding another level of indirection makes the code more complex and error prone and disable justify a benefit of a couple seconds extra compile time

Reply:
At the moment no but thats also what OpenRCT2 was like and then it slowly got worse and worse.

Reply:
I don't quite know how to nicely implement that change here then. I tried with the files under 'Windows' folder (see my last commit) but I don't think that does much? I'm not sure what exactly the fix should look like here.

Reply:
What you would do is have a windowstate.cpp file that acesses the gamestate and provide getters/setters into that. Then you only need to include the light weight  windowstate.h header in the window files and it will help group all of the windowstate into one area. Eventually that can become a struct in itself (can't reorg it nicely atm though) so it would be like:

```cpp
// WindowState.h
struct WindowState
{
    uint8_t lastTrackStyle;
    uint8_t lastBuildVehicleOption;
    ...
};

WindowState& getWindowState();
```

```cpp
// WindowState.cpp
#include "GameState.h"

WindowState& getWindowState(){
    return getGameState().windowState;
}
```

```cpp
// ShortcutManger.cpp
#include "WindowState.h"

if (getWindowState().lastBuildVehicleOption == x){
   doStuff
}
```

Reply:
Ah ok, that makes sense, thanks, I'll redo these PRs in this style